### PR TITLE
Allow updatedAt to be null to support legacy data

### PR DESCRIPTION
**Pull request by spackmat**
Fixing the BC break described in #496:  v1 had left `updatedAt` as `null`, if an entity was never updated. v2 introduces that `updatedAt` is initialized with the same timestamp as `createdAt`, so `updatedAt` cannot be `null` anymore. This PR changes the signature of `getUpdatedAt()` and `setUpdatedAt()` to allow null values again for legacy data (just to be consistent, as setting `updatedAt` to null will be overwritten on save).

### CODE REVIEW

**Comment by spackmat** [comment]
I didn't write a test for that because in the v2-world I cannot create a TimestampableEntity with `updatedAt` of `null`.

File: src/Contract/Entity/TimestampableInterface.php [hunk 1]
```diff
@@ -14,7 +14,7 @@ public function getCreatedAt(): DateTimeInterface;
 
     public function setCreatedAt(DateTimeInterface $createdAt): void;
 
-    public function getUpdatedAt(): DateTimeInterface;
+    public function getUpdatedAt(): ?DateTimeInterface;
 
-    public function setUpdatedAt(DateTimeInterface $updatedAt): void;
+    public function setUpdatedAt(?DateTimeInterface $updatedAt): void;
```
Comment (TomasVotruba) [inline src/Contract/Entity/TimestampableInterface.php]: Why this change?

Comment (spackmat) [reply src/Contract/Entity/TimestampableInterface.php]: Just for consistency of the API. Doesn't make sense as I wrote in the description. I can remove that again, but the whole PR is just a suboptimal workaround, see my new comment in the issue #496

**Comment by TomasVotruba** [comment]
I'm not sure what the problem is now. 
We need test to cover it.

In case of inconsistent date, I recommend using migration to complete them to database. We cannot compensate for broken data strutures here.

**Comment by spackmat** [comment]
Forget that PR at all. It is a workaround for a BC break that either has to fixed with a migration of existing data from v1 or by reverting the BC break and restoring the old behavior.
